types/policy_test.go

```diff
@@ -329,3 +333,123 @@ func TestPolicyRoundtrip(t *testing.T) {
 		t.Fatal("satisfied policy did not survive roundtrip:", sp, sp2)
 	}
 }
+
+func TestPolicyMarshaling(t *testing.T) {
+	privateKey := GeneratePrivateKey()
+	publicKey := privateKey.PublicKey()
+	publicKeyHex := hex.EncodeToString(publicKey[:])
+
+	// Generate a unique SHA256 hash from the current Unix time in nanoseconds for PolicyTypeHash test.
+	currentTime := time.Now().UnixNano()
+	currentTimeBytes := []byte(strconv.FormatInt(currentTime, 10))
+	hash := sha256.Sum256(currentTimeBytes)
+	hashHex := hex.EncodeToString(hash[:])
```

Comment:
these values don't need to be random; you can do:
```go
publicKey := NewPrivateKeyFromSeed(make([]byte, 32)).PublicKey()
hash := HashBytes(nil)
```
This ensures that the test will be deterministic, so the result can be cached.